client/controllers/map.js

```diff
@@ -1,8 +1,13 @@
 import { HTTP } from 'meteor/http';
 import { ReactiveVar } from 'meteor/reactive-var';
+import Locations from '/imports/collections/Locations';
 
 Template.map.onCreated(function () {
+  this.subscribe('locations');
```

Comment:
Subscribing to the full locations collection is going to take a lot of bandwidth. I recommend using a meteor method or parameterized subscription so that only the selected location is downloaded.